diagnostics-extension/src/dpsl.types.ts

```diff
@@ -0,0 +1,241 @@
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/**
+ * @fileoverview dpsl.* types definitions.
+ */
+
+// dpsl.telemetry.* type definitions
+
+/**
+ * Response message containing Backlight Info
+ */
+export interface BacklightInfoObject {
+  path: string;
+  maxBrightness: number;
+  brightness: number;
+}
+
+/**
+ * Response message containing Backlight Info
+ */
+export type BacklightInfo = BacklightInfoObject[];
+
+/**
+ * Response message containing Battery Info
+ */
+export interface BatteryInfo {
+  cycleCount: bigint;
+  voltageNow: number;
+  vendor: string;
+  serialNumber: string;
+  chargeFullDesign: number;
+  chargeFull: number;
+  voltageMinDesign: number;
+  modelName: string;
+  chargeNow: number;
+  currentNow: number;
+  technology: string;
+  status: string;
+  manufactureDate: string;
+  temperature: bigint;
+}
+
+/**
+ * Response message containing Bluetooth Info
+ */
+export interface BluetoothInfoObject {
+  name: string;
+  address: string;
+  powered: boolean;
+  numConnectedDevices: number;
+}
+
+/**
+ * Response message containing Bluetooth Info
+ */
+export type BluetoothInfo = BluetoothInfoObject[];
+
+/**
+ * Response message containing VPD Info
+ */
+export interface VpdInfo {
+  skuNumber: string;
+  serialNumber: string;
+  modelName: string;
+}
+
+/**
+ * Response message containing CPU Info
+ */
+export interface CpuInfo {
+  numTotalThreads: number;
+  architecture: string;
+  physicalCpus: Array<unknown>; // Replace unknown
+}
+
+/**
+ * Response message containing Fan Info
+ */
+export interface FanInfo {
+  speedRpm: number;
+}
+
+/**
+ * Response message containing Memory Info
+ */
+export interface MemoryInfo {
+  totalMemoryKib: number;
+  freeMemoryKib: number;
+  availableMemoryKib: number;
+  pageFaultsSinceLastBoot: bigint;
+}
+
+/**
+ * Response message containing BlockDevice Info
+ */
+export interface BlockDeviceInfoObject {
+  path: string;
+  size: bigint;
+  type: string;
+  manufacturerId: number;
+  name: string;
+  serial: string;
+  bytesReadSinceLastBoot: bigint;
+  bytesWrittenSinceLastBoot: bigint;
+  readTimeSecondsSinceLastBoot: bigint;
+  writeTimeSecondsSinceLastBoot: bigint;
+  ioTimeSecondsSinceLastBoot: bigint;
+  discardTimeSecondsSinceLastBoot: bigint;
+}
+
+/**
+ * Response message containing BlockDevice Info
+ */
+export type BlockDeviceInfo = BlockDeviceInfoObject[];
+
+/**
+ * Response message containing StatefulPartition Info
+ */
+export interface StatefulPartitionInfo {
+  availableSpace: bigint;
+  totalSpace: bigint;
+}
+
+/**
+ * Response message containing Timezone Info
+ */
+export interface TimezoneInfo {
+  posix: string;
+  region: string;
+}
+
+/**
+ * Union of the Telemetry Info types
+ */
+export type TelemetryInfoTypes =
+  | BacklightInfo
+  | BatteryInfo
+  | BluetoothInfo
+  | VpdInfo
+  | CpuInfo
+  | FanInfo
+  | MemoryInfo
+  | BlockDeviceInfo
+  | StatefulPartitionInfo
+  | TimezoneInfo;
```

Comment:
not sure if we need this. It was declared in dpsl/ for using inside the library code, not sure if the extension would need this?